pkg/fleet/installer/setup/djm/emr.go

```diff
@@ -0,0 +1,182 @@
+	IsMaster        bool   `json:"isMaster"`
+	InstanceGroupID string `json:"instanceGroupId"`
+}
+
+type cluster struct {
+	Name string `json:"Name"`
+}
+
+type emrResponse struct {
+	Cluster cluster `json:"cluster"`
+}
+
+type extraEmrInstanceInfo struct {
+	JobFlowID    string `json:"jobFlowId"`
+	ReleaseLabel string `json:"releaseLabel"`
+}
+
+// SetupEmr sets up the DJM environment on EMR
+func SetupEmr(s *common.Setup) error {
+
+	s.Packages.Install(common.DatadogAgentPackage, emrAgentVersion)
+	s.Packages.Install(common.DatadogAPMInjectPackage, emrInjectorVersion)
+	s.Packages.Install(common.DatadogAPMLibraryJavaPackage, emrJavaVersion)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("failed to get hostname: %w", err)
+	}
+	s.Config.DatadogYAML.Hostname = hostname
+	s.Config.DatadogYAML.DJM.Enabled = true
+
+	s.Config.DatadogYAML.ExpectedTagsDuration = "10m"
+	isMaster, clusterName, err := setupCommonEmrHostTags(s)
+	if err != nil {
+		return fmt.Errorf("failed to set tags: %w", err)
+	}
+	if isMaster {
+		setupEmrDriver(s, clusterName)
+	}
+	return nil
+}
+
+func setupCommonEmrHostTags(s *common.Setup) (bool, string, error) {
+
+	instanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "instance.json"))
+	if err != nil {
+		return false, "", fmt.Errorf("error reading instance file: %w", err)
+	}
+
+	var info emrInstanceInfo
+	if err = json.Unmarshal(instanceInfoRaw, &info); err != nil {
+		return false, "", fmt.Errorf("error umarshalling instance file: %w", err)
+	}
+
+	setHostTag(s, "instance_group_id", info.InstanceGroupID)
+	setHostTag(s, "is_master_node", strconv.FormatBool(info.IsMaster))
+	s.Span.SetTag("host_tag."+"is_master_node", info.IsMaster)
+
+	extraInstanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "extraInstanceData.json"))
+	if err != nil {
+		return info.IsMaster, "", fmt.Errorf("error reading extra instance data file: %w", err)
+	}
+
+	var extraInfo extraEmrInstanceInfo
+	if err = json.Unmarshal(extraInstanceInfoRaw, &extraInfo); err != nil {
+		return info.IsMaster, "", fmt.Errorf("error umarshalling extra instance data file: %w", err)
+	}
+	setHostTag(s, "job_flow_id", extraInfo.JobFlowID)
+	setHostTag(s, "cluster_id", extraInfo.JobFlowID)
+	s.Span.SetTag("emr_version", extraInfo.ReleaseLabel)
+
+	emrResponseRaw, err := executeCommandWithTimeout("aws", "emr", "describe-cluster", "--cluster-id", extraInfo.JobFlowID)
+	if err != nil {
+		return info.IsMaster, extraInfo.JobFlowID, fmt.Errorf("error describing emr cluster, using cluster id as name: %w", err)
+	}
+	var response emrResponse
+	if err = json.Unmarshal(emrResponseRaw, &response); err != nil {
+		return info.IsMaster, extraInfo.JobFlowID, fmt.Errorf("error unmarshalling AWS EMR response,  using cluster id as name: %w", err)
+	}
+
+	setHostTag(s, "cluster_name", response.Cluster.Name)
+	return info.IsMaster, response.Cluster.Name, nil
+}
+
+func setupEmrDriver(s *common.Setup, clusterName string) {
```

Comment:
nit: rename this variable to avoid confusion about the term driver. In YARN the master node is called a "ResourceManager" which is different than the spark driver (that can be running on worker nodes)

```suggestion
func setupEmrResourceManager(s *common.Setup, clusterName string) {
```